pkg/statistics/handle/storage/save.go

```diff
@@ -380,23 +381,33 @@ func SaveColOrIdxStatsToStorage(
 	return
 }
 
-// SaveMetaToStorage will save stats_meta to storage and update last stats histograms version.
+// SaveMetaToStorage will batch save stats_meta to storage.
 func SaveMetaToStorage(
 	sctx sessionctx.Context,
-	tableID, count, modifyCount int64,
 	refreshLastHistVer bool,
-) (statsVer uint64, err error) {
+	metaUpdates []statstypes.MetaUpdate,
+) (uint64, error) {
 	version, err := util.GetStartTS(sctx)
 	if err != nil {
 		return 0, errors.Trace(err)
 	}
+	var sql string
+	values := make([]string, 0, len(metaUpdates))
 	if refreshLastHistVer {
-		_, err = util.Exec(sctx, "replace into mysql.stats_meta (version, table_id, count, modify_count, last_stats_histograms_version) values (%?, %?, %?, %?, %?)", version, tableID, count, modifyCount, version)
```

Comment:
@winoros Is there any particular reason that we are using `replace into` here? It seems reasonable to change it to `insert into on delicate key update`.

Reply:
Also, please note that if we use the new approach(insert into on duplicate key update), the snapshot will remain as the old value. If we use replace into, the snapshot would be zero.